src/Storages/MergeTree/IExecutableTask.h

```diff
@@ -23,6 +23,7 @@ namespace DB
 class IExecutableTask
 {
 public:
+    using TaskResultCallback = std::function<void(bool)>;
```

Comment:
Not sure this should be in interface.
What this callback is needed for and when called?

Reply:
This callback is needed no assign another operation after current is finished. If current operation fails than we need to add a delay before to assign a new task.